### DISCUSSION TITLE
Newer versions of Mapperly on a Mac Developement environment

### DISCUSSION THREAD

**Discussion by zsolt3991** [riok/mapperly#1527] [Q&A]
Hello everyone,

Been using Mapperly for quite a while in projects, and I have hit somewhat of an intresting scenario. One of our projects got stuck on version 2.8.0 due to build issues in Visual Studio for Mac.

I am aware it has been recently fully deprecated by Microsoft, however it is still a better experience than VS Code where it cannot even correctly open Source Generated files.

The situation basically was that any newer versions like 3.x.x simply fail to be recognized in the IDE as Source Generators, throwing errors for all mapping methods for being a partial method without implementation. It will build in CLI just fine but the IDE still stays messed up.

So I am wondering what IDEs you guys are using? VS on Windows seems fine with any version, on Mac only Rider seems to be somewhat competent with all versions, even though it seems to not show the Analyzer output from the Mapperly analyzers which is somewhat weird.

**Comment by latonz** [comment]
I‘m using Rider for working on Mapperly as well as when using Mapperly in a project. Works fine for me. For the issue with VS Vode: have you tried setting `EmitCompilerGeneratedFiles` see [docs](https://mapperly.riok.app/docs/configuration/generated-source/)?

**Comment by zsolt3991** [comment]
You are right. `EmitCompilerGeneratedFiles` has worked for me previously in VS Code to view the Generator output. Its just the navigation to references that is cumbersome, but that seems a VSCode issue in general.

Currently I am also using Rider, was just wondering if there is any workaround for making it work in VS for Mac as well.
